src/shout_subgroup/repository.py

```diff
@@ -110,6 +111,43 @@ async def insert_subgroup(
     return new_subgroup
 
 
+async def delete_subgroup(
+        db: Session,
+        telegram_group_chat_id: int,
+        subgroup_name: str,
+) -> bool:
+    """
+    Hard deletes a subgroup from the database.
+    This delete is a cascading delete, meaning
+    if there are users associated with the subgroup,
+    the associations will be removed prior to the deletion.
+    :param db:
+    :param telegram_group_chat_id:
+    :param subgroup_name:
+    :return:
+    """
+
+    subgroup = await find_subgroup_by_telegram_group_chat_id_and_subgroup_name(
+        db,
+        telegram_group_chat_id,
+        subgroup_name
+    )
+
+    if not subgroup:
+        return False
+
+    # This is a cascading delete.
+    # We'll remove the users in the subgroup prior to deletion
+    subgroup.users.clear()
+    db.commit()
+
+    # Delete the subgroup
+    db.delete(subgroup)
+    db.commit()
+
+    return True
+
+
 async def insert_group_chat(db: Session, telegram_chat: Chat) -> GroupChatModel:
     new_group_chat = GroupChatModel(
         telegram_group_chat_id=telegram_chat.id,
```

Comment:
Let's discuss on Monday.

In short, I view the models as objects that represent database structure. Calling functions shouldn't need to know the underlying representation. From the callers point of view, they pass in primitive arguments and are returned a model.

But I understand the argument of, "since the model is being returned, the client already knows about it". 

We can decide where to draw the line in sand of responsibility. Then we'll refactor to keep it consistent throughout the code base.

Reply:
Created issue #37 to address this